src/components/Modal/Modal.js

```diff
@@ -75,79 +84,83 @@ class Modal extends Component {
+      ariaDescribedby,
+      dataId,
+      children,
+      hasCloseButton,
       ...props
     } = this.props
 
     return (
-      <div ref={this.mainRef}>
-        {/* needed for close event from crashing. may need a better solution */}
-        <div data-sprk-modal-trigger={id} />
-        <div
-          className={this.className}
-          ref={this.modalRef}
-          role='dialog'
-          tabIndex='-1'
-          aria-labelledby={ariaLabelledby}
-          aria-modal='true'
-          aria-describedby={ariaDescribedby}
-          data-sprk-modal={id}
-          data-id={dataId}
-          {...props}
-        >
-          <div className='sprk-o-Stack sprk-o-Stack--large'>
-            {type === MODAL_VARIANTS.INFO && (
-              <InfoContent
-                modalName={id}
-                body={children}
-                close={() => this.hide()}
-              />
-            )}
-            {type === MODAL_VARIANTS.CHOICE && (
-              <ChoiceContent
-                modalName={id}
-                body={children}
-                close={() => this.hide()}
-              />
-            )}
-            {type === MODAL_VARIANTS.WAIT && (
-              <WaitContent
+      <div className='sprk-u-JavaScript'>
+        <div data-sprk-modal-trigger={id} />  {/* This needs to be fixed, their method looks for this attribute and sets focus to it */}
+        <div ref={this.mainRef}>
+          <div
+            className={this.className}
+            ref={this.modalRef}
+            role='dialog'
+            tabIndex='-1'
+            aria-labelledby={ariaLabelledby}
+            aria-modal='true'
+            aria-describedby={ariaDescribedby}
+            data-sprk-modal={id}
+            data-sprk-modal-type={type}
+            data-id={dataId}
+            {...props}
+          >
+            <Stack itemSpacing={'large'}>
+              <ModalHeader
+                id={ariaLabelledby}
+                onClose={onClose}
+                title={title}
                 modalName={id}
-                body={children}
-                close={() => this.hide()}
+                hasCloseButton={type !== MODAL_VARIANTS.WAIT ? hasCloseButton : false}
               />
-            )}
+              <ModalBody>
+                {type === MODAL_VARIANTS.WAIT && <Spinner />}
+                <p className='sprk-o-Stack__item sprk-b-TypeBodyTwo' id={ariaDescribedby}>
+                  {children}
+                </p>
+              </ModalBody>
+              {
+                type === MODAL_VARIANTS.CHOICE &&
+                <ModalFooter
+                  modalId={id}
+                  onConfirm={onConfirm}
+                  onCancel={onCancel}
+                  confirmText={confirmText}
+                  cancelText={cancelText}
+                />
+              }
+            </Stack>
           </div>
+          <div
+            data-sprk-modal-mask='true'
+            className='sprk-c-ModalMask sprk-u-Display--none'
```

Comment:
same here with the class name, we should be generating it, not hard coding in case they ever change

Reply:
I just made a ModalMask component